src/nemo_evaluator/benchmarks/pinchbench.py

```
@@ -70,13 +79,27 @@ def _cleanup_workspaces() -> None:
+    *,
+    source_ref: str = _DEFAULT_PINCHBENCH_SOURCE_REF,
+    source_url: str = _DEFAULT_PINCHBENCH_SOURCE_URL,
+) -> Path:
+    key = _cache_key(source_ref, source_url)
+    cached = _CACHE_DIRS.get(key)
+    if cached is not None and cached.exists():
+        return cached
+
+    archive_url = _archive_url(source_url, source_ref)
+    logger.info("Downloading PinchBench tasks from %s", archive_url)
+    with urllib.request.urlopen(archive_url, timeout=120) as resp:
```

Ruff (0.15.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)